File: go/space-age/space_age.go
// Package space calculates person's age in years related to different planets
package space

type Planet string

var orbitalPeriods = map[Planet]float64{
	"Earth":   1.0,
	"Jupiter": 11.862615,
	"Mars":    1.8808158,
	"Mercury": 0.2408467,
	"Neptune": 164.79132,
	"Saturn":  29.447498,
	"Uranus":  84.016846,
	"Venus":   0.61519726,
}

const secondsInYear = 31557600.0

// Age calculates person's age in years from given seconds and planet
func Age(s float64, p Planet) float64 {
	return s / orbitalPeriods[p] / secondsInYear
}

File: typescript/space-age/space-age.ts
enum PlanetRatios {
  Earth = 31557600,
  Mercury = 0.2408467 * Earth,
  Venus = 0.61519726 * Earth,
  Mars = 1.8808158 * Earth,
  Jupiter = 11.862615 * Earth,
  Saturn = 29.447498 * Earth,
  Uranus = 84.016846 * Earth,
  Neptune = 164.79132 * Earth,
}

export default class SpaceAge {
  public seconds: number;

  constructor(seconds: number) {
    this.seconds = seconds;
  }

  onEarth = (): number => this.toYear(PlanetRatios.Earth);
  onMercury = (): number => this.toYear(PlanetRatios.Mercury);
  onVenus = (): number => this.toYear(PlanetRatios.Venus);
  onMars = (): number => this.toYear(PlanetRatios.Mars);
  onJupiter = (): number => this.toYear(PlanetRatios.Jupiter);
  onSaturn = (): number => this.toYear(PlanetRatios.Saturn);
  onUranus = (): number => this.toYear(PlanetRatios.Uranus);
  onNeptune = (): number => this.toYear(PlanetRatios.Neptune);

  private toYear(ratio: number): number {
    return Number((this.seconds / ratio).toFixed(2));
  }
}
